### ISSUE TITLE
There's no difference between "search box" and "dropdown list" in a dashboard filter

### ISSUE THREAD

**Issue by dragonsahead** [metabase/metabase#31656]
### Describe the bug

When selecting the type of filter you would like in a categorical value, the dropdown list and search box have the same functionality

### To Reproduce

1) create a gui question
2) create a dashboard
3) add a category filter to the dashboard
4) select search box or dropdown list and check what you see, both will act the same

### Expected behavior

We shouldn't suggest values when there's a search box

### Logs

NA

### Information about your Metabase installation

```JSON
v1.46.5
```

### Severity

P3

**Comment by uladzimirdev** [comment]
put on hold as it's not clear how to address the issue @paoliniluis any idea from your side?

**Comment by uladzimirdev** [comment]
To reproduce the issue you need to select `From connected fields` and choose a question where Category type is available (e.g. Product.Category).

![image](https://github.com/metabase/metabase/assets/125459446/bb837dab-b36e-4d69-99e9-8e21d63bf52a)


from discussion with the team

`Search Box` is different from `Dropdown List`:
- `Search box` doesn't show all the values in the list by default, instead it provides an input field and asks for values once user types something
- `Search box` is a default variant for the question with more than 1k of values

Wrong behaviour of the Search Box described in the issue is most likely related to the case, when all values are displayed in Search Box with checkboxes (multiple selection is enabled), when there shouldn't be any value by default

So, no difference between Search Box and Dropdown
| Search Box | Dropdown |
|--------|--------|
| <img width="464" alt="image" src="https://github.com/metabase/metabase/assets/125459446/38b1a186-6604-4b43-b4fa-fcd18ec637a3"> | <img width="647" alt="image" src="https://github.com/metabase/metabase/assets/125459446/45f4007f-35ee-4a34-a94e-b52248bb4a6b">
 |

**Comment by cdeweyx** [comment]
Moving to "Soon" as the solution here isn't clear atm and it's a P3

**Comment by mekedron** [comment]
Any updates on this one? It can be reproduced on [v0.57.3.3](https://hub.docker.com/layers/metabase/metabase/v0.57.3.3/images/sha256-9fbd66aaf274883f7ed395d6eaa050513f9e7ed670cfdbe8fce692490a1d024a) and [v0.57.4.2](https://hub.docker.com/layers/metabase/metabase/v0.57.4.2/images/sha256-2b3332574ced1a30ebf6a9aa840067ef5ff5ddcdef697fc63f95d684fbaadbf9)

<img width="1629" height="968" alt="Image" src="https://github.com/user-attachments/assets/d1ffeb1f-da48-41e9-af19-67da5e6cd518" />

**Comment by dragonsahead** [comment]
@mekedron how many distinct items does the column have?

**Comment by mekedron** [comment]
Hey @paoliniluis

sessionSource has 254 distinct items, and sessionMedium has 38, and both fields have this issue

And this search field is super inconvinient when multiple items need to be selected. After choosing an item, I need to perform search again, and again, and again
![Image](https://github.com/user-attachments/assets/6eafe497-b064-4411-9f47-04934c4abee2)

**Comment by mekedron** [comment]
UPD:

Ah, my bad! I didn't know that each field has it's own setting!

<img width="1667" height="964" alt="Image" src="https://github.com/user-attachments/assets/8ffbbbc7-ac7b-43e7-95af-b440cd0da147" />


Now everything works as expected:

<img width="374" height="322" alt="Image" src="https://github.com/user-attachments/assets/34eb5b9b-902a-4cf6-858a-4e0c3415fcb2" />

**Comment by ranquild** [comment]
Yes, this is a known limitation - we don't "override" the underlying field setting.
